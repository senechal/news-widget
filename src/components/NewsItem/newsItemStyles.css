.news-item{
  border-bottom: 1px solid #cecece;
  padding: 15px 0px;
  vertical-align: middle;
  width: 100%;
}

.news-item-title{
  font-weight: 700;
  font-size: 16px;
  color: #000;
  text-decoration: none;
  display: block;
  margin-bottom: 12px;
}
.news-item-title:hover{
  text-decoration: underline;
}

.news-item-date{
  display: inline;
  color: #000;
  font-size: 12px;
  margin-right: 15px;
}

.news-item-source{
  font-size: 10px;
  color: #fff;
  background-color: #cecece;
  border-radius: 5px;
  padding: 2px 10px;
  display: inline;


}