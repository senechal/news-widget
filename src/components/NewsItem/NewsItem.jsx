import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import './newsItemStyles.css';

const NewsItem = ({
  title, url, date, source,
}) => (
  <div className="news-item">
    <a className="news-item-title" href={url}>
      {title}
    </a>
    <div className="news-item-date">{moment(date).format('DD/MM/YYYY')}</div>
    <div className="news-item-source">{source.name}</div>
  </div>
);

NewsItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  source: PropTypes.objectOf(PropTypes.string).isRequired,
  date: PropTypes.string.isRequired,
};

export default NewsItem;
