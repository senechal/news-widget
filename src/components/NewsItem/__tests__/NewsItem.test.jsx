import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import NewsItem from '../NewsItem';

describe('Test for NewsItemt', () => {
  configure({ adapter: new Adapter() });
  it('should render correctly', () => {
    const wrapper = shallow(<NewsItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
