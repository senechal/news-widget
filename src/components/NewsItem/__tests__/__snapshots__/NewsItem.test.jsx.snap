// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for NewsItemt should render correctly 1`] = `
<div>
  News Item
</div>
`;
